### PR TITLE
SC-24

### PR DESCRIPTION
**Pull request by ahayden**
This statement should join strings to make a valid policy... but I'm not certain.

Passes pre-commit

### CODE REVIEW

File: s3/sc-s3-encrypted-ra.yaml [hunk 1]
```diff
@@ -41,7 +41,13 @@ Resources:
...
             Effect: "Allow"
             Principal:
-              AWS: !Ref 'S3UserARNs'
+              !Join
+                - ""
+                  - "AWS "
+                  - !Join
+                    - ","
+                      - !Ref ${S3UserARNs}
+                      - "arn:aws:iam::325565585839:root"
```
Comment (tthyer) [inline s3/sc-s3-encrypted-ra.yaml]: SC-24 concerns the synapse bucket policy, not this bucket template, so please undo these changes

File: s3/sc-s3-synapse-ra.yaml [hunk 2]
```diff
@@ -58,7 +58,13 @@ Resources:
...
             Effect: "Allow"
             Principal:
-              AWS: !Ref 'S3UserARNs'
+              !Join
+                - ""
+                  - "AWS "
+                  - !Join
+                    - ","
+                      - !Ref ${S3UserARNs}
+                      - "arn:aws:iam::325565585839:root"
```
Comment (tthyer) [inline s3/sc-s3-synapse-ra.yaml]: This looks odd. I don't think you need two joins. Don't need the `- "AWS"`. Does this template validate? I.e. if you run `aws cloudformation validate-template --template-body <path-to-your-file>`?

File: s3/sc-s3-synapse-ra.yaml [hunk 3]
```diff
@@ -26,7 +26,7 @@ Parameters:
     Default: ""
 Conditions:
   HasBucketName: !Not [!Equals [!Ref BucketName, ""]]
-  AllowUserAccess: !Not [!Equals [!Join ['', !Ref S3UserARNs], "[]"]]
+  AllowUserAccess: !Not [!Equals [!Join ['', !Ref S3UserARNs], "[]"]]
```
Comment (zaro0508) [inline s3/sc-s3-synapse-ra.yaml]: undo this

File: s3/sc-s3-synapse-ra.yaml [hunk 4]
```diff
@@ -45,7 +45,7 @@ Resources:
             MaxAge: 3000
       Tags:
         - Key: Name
-          Value: SC-S3-RA-S3-Bucket
+          Value: SC-S3-RA-S3-Bucket
```
Comment (zaro0508) [inline s3/sc-s3-synapse-ra.yaml]: undo this

File: s3/sc-s3-synapse-ra.yaml [hunk 5]
```diff
@@ -67,7 +73,7 @@ Resources:
             Sid: "WriteAccess"
             Effect: "Allow"
             Principal:
-              AWS: !Ref 'S3UserARNs'
+              AWS: !Ref 'S3UserARNs' #sc-requestor write access
```
Comment (zaro0508) [inline s3/sc-s3-synapse-ra.yaml]: you'll need to do the same thing with `WriteAccess` as you did with `ReadAccess` because we want the ARN to have both read/list and write access.

File: s3/sc-s3-synapse-ra.yaml (hunk 2 above)
Comment (tthyer) [reply s3/sc-s3-synapse-ra.yaml]: To be more clear, it currently looks like this:
```
Principal:
    AWS: !Ref 'S3UserARNs'
```
and I think whatever you're going to do should look something like this:
```
Principal:
    AWS: !Join ...
```
Join is going to create a string. Whatever is on the right side of "AWS:" needs to be a list. So it's possible that after combining the root ARN with the S3UserARNs list you'll have to Fn::Split.

Comment (zaro0508) [reply s3/sc-s3-synapse-ra.yaml]: maybe you can try this?
```
Principal: !Join
        - ','
        - 'arn:aws:iam::325565585839:root'
        - !Ref S3UserARNs
```

File: s3/sc-s3-synapse-ra.yaml (hunk 5 above)
Comment (ahayden) [reply s3/sc-s3-synapse-ra.yaml]: This was one open question... Should the Synapse prod account really have write access to this bucket?

Comment (ahayden) [reply s3/sc-s3-synapse-ra.yaml]: This was one open question... Should the Synapse prod account really have write access to this bucket?

File: s3/sc-s3-synapse-ra.yaml (hunk 2 above)
Comment (ahayden) [reply s3/sc-s3-synapse-ra.yaml]: The validation tool returns errors, but also errors with the file as it is in the master branch...

Comment (ahayden) [reply s3/sc-s3-synapse-ra.yaml]: This new commit validates OK

File: s3/sc-s3-synapse-ra.yaml [hunk 6]
```diff
@@ -25,8 +25,9 @@ Parameters:
     Description: (Optional) Name of the created bucket.
     Default: ""
 Conditions:
-  HasBucketName: !Not [!Equals [!Ref BucketName, ""]]
+  HasSynapseUserName: !Not [!Equals [!Ref SynapseUserName, ""]]
```
Comment (zaro0508) [inline s3/sc-s3-synapse-ra.yaml]: This is a good thought but I don't think it's necessary.  The purpose of this template is to create a synapse bucket product so a SynapseUserName is required otherwise it won't be a synapse bucket.  

We should also remove the `default: ""` from properties section.

File: s3/sc-s3-synapse-ra.yaml [hunk 7]
```diff
@@ -25,8 +25,9 @@ Parameters:
...
   AllowUserAccess: !Not [!Equals [!Join ['', !Ref S3UserARNs], "[]"]]
+  HasBucketName: !Not [!Equals [!Ref BucketName, ""]]
```
Comment (zaro0508) [inline s3/sc-s3-synapse-ra.yaml]: I don't see this being used anywhere?

File: s3/sc-s3-synapse-ra.yaml [hunk 8]
```diff
@@ -58,7 +59,11 @@ Resources:
             Sid: "ReadAccess"
             Effect: "Allow"
             Principal:
-              AWS: !Ref 'S3UserARNs'
+              AWS: !Split
+                - ","
+                - !Sub
+                  - "${list},arn:aws:iam::325565585839:root"
```
Comment (zaro0508) [inline s3/sc-s3-synapse-ra.yaml]: would it be possible to parameterize the synapse ARN and then ref it here like so..
```
- "${list},${SynapsAccountArn}"
```
That will allow us to easily pass in a different ARN in case it changes for some reason.

File: s3/sc-s3-synapse-ra.yaml [hunk 9]
```diff
@@ -58,7 +59,11 @@ Resources:
             Sid: "ReadAccess"
             Effect: "Allow"
             Principal:
-              AWS: !Ref 'S3UserARNs'
+              AWS: !Split
```
Comment (zaro0508) [inline s3/sc-s3-synapse-ra.yaml]: nice job finding this wacky yaml 👍

File: s3/sc-s3-synapse-ra.yaml (hunk 8 above)
Comment (tthyer) [reply s3/sc-s3-synapse-ra.yaml]: So you want that to be a Parameter? I see your point, but that's another thing for the user to enter so opportunity for more user error. Could it be a Ref?

File: s3/sc-s3-synapse-ra.yaml [hunk 10]
```diff
@@ -73,7 +72,11 @@ Resources:
             Sid: "WriteAccess"
             Effect: "Allow"
             Principal:
-              AWS: !Ref 'S3UserARNs' #sc-requestor write access
+              AWS: !Split
+                - ","
+                - !Sub
+                  - "${list},arn:aws:iam::325565585839:root" 
+                  - list: !Join [",",!Ref "S3UserARNs"]
```
Comment (tthyer) [inline s3/sc-s3-synapse-ra.yaml]: @ahayden I haven't ever seen this syntax before `list: ` -- do you have a reference for it? I can't find anything.

Comment (zaro0508) [reply s3/sc-s3-synapse-ra.yaml]: wacko! https://bl.ocks.org/magnetikonline/a6cfc522a1e9f876b75962f5f553c8e5

File: s3/sc-s3-synapse-ra.yaml (hunk 8 above)
Comment (zaro0508) [reply s3/sc-s3-synapse-ra.yaml]: SC will always show the param even when there's a default?  can it be hidden with adjustments to `AWS::CloudFormation::Interface` section?

File: s3/sc-s3-synapse-ra.yaml (hunk 6 above)
Comment (ahayden) [reply s3/sc-s3-synapse-ra.yaml]: fair enough... wanting to avoid future bugs

File: s3/sc-s3-synapse-ra.yaml (hunk 10 above)
Comment (ahayden) [reply s3/sc-s3-synapse-ra.yaml]: I saw it in this gist... https://gist.github.com/magnetikonline/a6cfc522a1e9f876b75962f5f553c8e5

File: s3/sc-s3-synapse-ra.yaml (hunk 8 above)
Comment (ahayden) [reply s3/sc-s3-synapse-ra.yaml]: Agreed- I'll define a simple parameter at the top that can be overridden from the default if needed

File: s3/sc-s3-synapse-ra.yaml (hunk 7 above)
Comment (ahayden) [reply s3/sc-s3-synapse-ra.yaml]: This is checked a few lines later
